bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryService.java

```diff
@@ -68,18 +68,21 @@ public class ThingDiscoveryService extends AbstractDiscoveryService implements T
     protected static final Map<String, ThingTypeUID> DEVICEMODEL_TO_THINGTYPE_MAP = Map.ofEntries(
             new AbstractMap.SimpleEntry<>("BBL", BoschSHCBindingConstants.THING_TYPE_SHUTTER_CONTROL),
             new AbstractMap.SimpleEntry<>("TWINGUARD", BoschSHCBindingConstants.THING_TYPE_TWINGUARD),
-            new AbstractMap.SimpleEntry<>("PSM", BoschSHCBindingConstants.THING_TYPE_INWALL_SWITCH),
+            new AbstractMap.SimpleEntry<>("BSM", BoschSHCBindingConstants.THING_TYPE_INWALL_SWITCH),
+            new AbstractMap.SimpleEntry<>("PSM", BoschSHCBindingConstants.THING_TYPE_SMART_PLUG_COMPACT),
             new AbstractMap.SimpleEntry<>("PLUG_COMPACT", BoschSHCBindingConstants.THING_TYPE_SMART_PLUG_COMPACT),
             new AbstractMap.SimpleEntry<>("CAMERA_360", BoschSHCBindingConstants.THING_TYPE_CAMERA_360),
             new AbstractMap.SimpleEntry<>("CAMERA_EYES", BoschSHCBindingConstants.THING_TYPE_CAMERA_EYES),
-            new AbstractMap.SimpleEntry<>("BWTH", BoschSHCBindingConstants.THING_TYPE_THERMOSTAT), // wall thermostat
+            new AbstractMap.SimpleEntry<>("BWTH", BoschSHCBindingConstants.THING_TYPE_WALL_THERMOSTAT), // wall thermostat
```

Comment:
Regarding mapping:
1. its hard to map the right KeyStrings like "BWTH", because of the lack of good docu. But I have no other idea as (reverse) engineering and user feedback
2. mapping a device to the right services is easier. The SHC API offers a list of services and we need to ensure that our openhab devices support all possible services. Therefore this mapping should be possible based on interface documentation and if the mapping is e.g. based on names we can check or do the mapping via code.

Both points could be discussed and improved in later PRs. This PR improves the code and shall be merged!

Reply:
> Mapping BWTH to THING_TYPE_WALL_THERMOSTAT results in a BatteryLevel Channel/Item and no value can be received. This makes sense for me, because my thermostats are 230V powered.

I discovered this during test, but the main code results in the same.